src/server/server.cc

```diff
@@ -1683,12 +1683,21 @@ void Server::UpdateWatchedKeysFromArgs(const std::vector<std::string> &args, con
   if (attr.IsWrite() && watched_key_size_ > 0) {
     if (attr.key_range.first_key > 0) {
       updateWatchedKeysFromRange(args, attr.key_range);
-    } else if (attr.key_range.first_key < 0) {
+    } else if (attr.key_range.first_key == -1) {
       redis::CommandKeyRange range = attr.key_range_gen(args);
 
       if (range.first_key > 0) {
         updateWatchedKeysFromRange(args, range);
       }
+    } else if (attr.key_range.first_key == -2) {
+      std::vector<redis::CommandKeyRange> vec_range = attr.key_range_vec_gen(args);
+
+      if (vec_range.front().first_key > 0) {
```

Comment:
The program will abort if `vec_range.size() == 0`. We can check every range in the for loop instead.